src/zarr/core/metadata/v2.py

```diff
@@ -165,18 +157,44 @@ def update_attributes(self, attributes: dict[str, JSON]) -> Self:
         return replace(self, attributes=attributes)
 
 
-def parse_zarr_format(data: Literal[2]) -> Literal[2]:
+def parse_zarr_format(data: object) -> Literal[2]:
     if data == 2:
-        return data
+        return 2
     raise ValueError(f"Invalid value. Expected 2. Got {data}.")
 
 
-def parse_filters(data: list[dict[str, JSON]] | None) -> list[dict[str, JSON]] | None:
-    return data
+def parse_filters(data: object) -> tuple[numcodecs.abc.Codec, ...] | None:
```

Comment:
```suggestion
def parse_filters(data: None, Iterable[numcodecs.abc.Codec, dict]) -> tuple[numcodecs.abc.Codec, ...] | None:
```

Reply:
similar answer to https://github.com/zarr-developers/zarr-python/pull/2179#discussion_r1758393646: these parsing functions take arbitrary python objects by design. type checking errors for the inputs to these functions don't help with that.